pkg/distributor/distributor.go

```diff
@@ -1056,6 +1063,9 @@ func (d *Distributor) limitsMiddleware(next push.Func) push.Func {
 			return nil, httpgrpc.Errorf(http.StatusTooManyRequests, validation.NewRequestRateLimitedError(d.limits.RequestRate(userID), d.limits.RequestBurstSize(userID)).Error())
 		}
 
+		// Note that we don't enforce the per-user ingestion rate limit here since we need to apply validation
+		// before we know how many samples/exemplars/metadata we'll actually be writing.
+
```

Comment:
Is this related to this PR?

Reply:
Only in the sense that it's about limits.